### DISCUSSION TITLE
How to use computed columns with Alerting?

### DISCUSSION THREAD

**Discussion by greatvovan** [grafana/grafana-infinity-datasource#484] [Q&A] [answered]
Hello,

I have the following JSON from an API:

```json
{
  "water_licenses": [
    {
      "total_sum": 16.804525985522538,
      "license_id": "C00001",
      "purpose": "AAA",
      "limit": 5184.0,
      "over_limit": false
    },
    {
      "total_sum": 0.0,
      "license_id": "C00002",
      "purpose": "BBB",
      "limit": 30000.0,
      "over_limit": false
    },
    {
      "total_sum": 18.964691922839016,
      "license_id": "C00003",
      "purpose": "CCC",
      "limit": 50026.0,
      "over_limit": false
    }
  ]
}
```

I need to calculate a percentage column as `total_sum` divided by `limit`.

In Explore mode, I can easily calculate that as a computed column:

<img width="1697" alt="image" src="https://user-images.githubusercontent.com/4903007/205850718-02e3a169-829e-4738-975b-dc49d2dfabcd.png">

But Alerting requires to have only a single numeric column, otherwise it complains:

<img width="1641" alt="image" src="https://user-images.githubusercontent.com/4903007/205851675-8a61a544-5011-4eb5-a39f-58a036895687.png">

How can I remove `Limit` and `Total` columns after computation?

**Comment by yesoreyeram** [comment, marked as answer]
I don't have a direct answer. But summarize should do the trick for you.

**Reply by greatvovan** [reply]
Thanks, summarize worked.

On a side note, is it possible to make UQL working on the Back End? It is the most powerful and flexible language allowing us to achieve almost any type of data manipulation and avoid workarounds like this. Would be highly appreciated!

**Reply by yesoreyeram** [reply]
Unfortunately UQL is currently javascript Library. There is no golang version of it yet.

**Reply by greatvovan** [reply]
Quick googling brought me to these high-starred projects:

https://github.com/robertkrimen/otto
https://github.com/dop251/goja

Maybe you can run UQL in one of these interpreters? Who knows when Golang version appears. Running the original JS version will make it easier to integrate the latest updates of the language.

**Reply by yesoreyeram** [reply]
looks interesting. Anyhow, It is not in the priority list yet. Feel free to PR.
